ARCs/arc-0004.md

```diff
@@ -143,12 +143,31 @@ interface Method {
     /** Optional, user-friendly description for the argument */
     desc?: string
   }>,
+  /** A list of allowed OnCompletion values for the method. Must not be empty */
+  allowedActions: OnCompletionAction[],
   /** Information about the method's return value */
   returns: { type: string, desc?: string }
 }
 ```
 
-For example:
+Where `OnCompletionAction` is defined as:
+
+```typescript
+type OnCompletionAction = "noop" | "optin" | "closeout" | "clearstate" | "update" | "delete";
```

Comment:
Should we switch to the TEAL constants (https://developer.algorand.org/docs/get-details/dapps/avm/teal/specification/#oncomplete):
`NoOp`, `OptIn`, `CloseOut`, `ClearState`, `UpdateApplication`, `DeleteApplication`
for consistency?

Reply:
It seems weird to have clearstate here.  Contracts can't say no.  They could return 0, which I suppose would cause the effects not to occur, but the clearstate would still happen.